### PR TITLE
Update agent-tools docs: retire create-organization, mark local MCP experimental

### PR DESCRIPTION
**Pull request by bryanfriedman**
## What

Brings the agent-tools docs in line with what the CLI actually ships.

This PR started as the docs counterpart to moderneinc/moderne-cli#4393, which was going to restore both `create-recipe` and `create-organization`. **That PR changed before merging: `create-organization` stayed retired and only `create-recipe` came back.** The scope here grew accordingly — see [Why this grew](#why-this-grew) below.

## Why this grew

Three things pushed this past its original scope:

1. **`create-organization` is removed as well.** Every reference had to go, not just be updated. It was also load-bearing in the AI recipes workshop, which was built around the loop that `create-recipe` and `create-organization` formed together.
2. **The AI recipes workshop stopped making sense skills-first.** Rather than pull the course, it now leads with the development loop itself and treats `create-recipe` as an optional accelerator. Coding models already handle OpenRewrite reasonably well, since the framework is open source and well represented in training data, so the exercises work with the agent and the CLI alone.
3. **The local MCP server needed an experimental label**, which touches the same pages.

## Changes

**Skills page**

- Drops `create-organization` entirely. The standalone group is now `create-recipe` and `prethink`.
- Regroups the sections as "skills that call no MCP tool" rather than "work with the CLI alone". `prethink` calls no MCP tool, but it is not self-contained: it reads `.moderne/context/`, which only exists once the Prethink recipe has run. The old grouping implied install-and-go.
- Removes the workflow example. With `create-organization` gone, three of its four steps were plain CLI commands with one skill mention wedged in.
- Each MCP-backed skill still names the tool(s) it fronts, verified against the `See the <skill> skill` markers in the tool descriptions rather than inferred from names.

**Local MCP server marked experimental**

- A warning on the four local-server pages (`overview`, `getting-started`, `tool-browser`, `security`) and the two workshop modules that use it.
- The **remote** server is unaffected and deliberately unmarked.
- The full wording lives on the MCP overview only; every other page points at it, so the language is maintained in one place.
- The security page keeps a reviewer-specific line: re-check it against the CLI version being approved rather than treating the tool list as fixed.

**AI recipes workshop**

- Exercise 1-1 now sets up the agent and introduces the loop, instead of introducing the skills. Installing skills is an optional step.
- The `create-organization` repo-discovery path is replaced with a direct prompt, including the instruction to confirm builds and tests pass before a repo joins the test set. That matters because the workshop judges the recipe by whether repos still compile afterward, which is a worthless signal if they were already broken.
- Drops the "the skill does X" attributions in Modules 2 and 3, which credited a skill the reader may not have installed.

**Other**

- Removes the duplicated skill roster from the agent tools install module and links to the skills page, so the list lives in one place.

### CODE REVIEW

**Review by mike-solomon — APPROVED** [review]
I cleaned up some more things I thought were wrong - if you could take a look at the changes and make sure they make sense @bryanfriedman 

Otherwise LGTM. Unsure when we want to merge this as I see the CLI PR looks like it's not getting put in for a bit.

File: docs/user-documentation/agent-tools/mcp/overview.md [hunk 1]
```diff
@@ -64,13 +68,14 @@ The MCP server exposes the following tools:
...
+| `pattern_replace`    | Compiles and runs a Kotlin recipe-DSL source (the `rewrite { } to { }` shape) to make structural code changes across the codebase. Use when `edit_code` finds no fitting marketplace recipe.                                                                            |
 
 ### Working with recipes
 
 | Tool              | Description                                                                                                                              |
 |-------------------|------------------------------------------------------------------------------------------------------------------------------------------|
-| `search_recipes`  | Searches available OpenRewrite recipes by natural-language query. Returns a paginated list of matching recipe names ranked by relevance. |
+| `edit_code`       | Finds a curated marketplace recipe that transforms code across the repository. First step, before `learn_recipe` and `run_recipe`.       |
+| `analyze_code`    | Finds a curated recipe that analyzes code across the repository without modifying it. Feeds `query_datatable` for structured results.    |
```
[the remark this answers is not in the document]
Comment (bryanfriedman) [reply docs/user-documentation/agent-tools/mcp/overview.md]: I think the idea is that the tool names reflect jobs that the agent wants to be able to do?

File: docs/hands-on-learning/ai-recipes/module-3-test.md [hunk 2]
```diff
@@ -38,61 +38,76 @@ mod git sync csv working-set working-set/repos.csv --with-sources
...
-The skill handles the full workflow: compiling your recipe, searching the working set for target patterns (pre-analysis), running the recipe, comparing results to predictions, and reporting findings. This will take several minutes. Watch as it goes.
+For reference, the underlying commands are:
+
+```bash
+./gradlew publishToMavenLocal
+mod config recipes jar install <groupId>:<artifactId>:<version>
+mod config recipes active set <RECIPE_PATH>
+mod build working-set --no-download --streaming
+mod run working-set --active-recipe --streaming --parallel 2
+```
```
[the remark this answers is not in the document]
Comment (bryanfriedman) [reply docs/hands-on-learning/ai-recipes/module-3-test.md]: Yes, the skill does include that command, but we can change it here at least for now.

**Review by mike-solomon — APPROVED** [review]
Changes look good to me other than what Sam mentioned of dropping the no download flag. Thanks for your work on this!
